sql/parser/builtins.go

```diff
@@ -72,6 +74,55 @@ var builtins = map[string]builtin{
 	"upper": stringBuiltin(func(s string) (Datum, error) {
 		return DString(strings.ToUpper(s)), nil
 	}),
+
+	"substr": {
+		nArgs: 3,
+		fn: func(args DTuple) (Datum, error) {
+			dstr, ok := args[0].(DString)
+			if !ok {
+				return DNull, argTypeError(args[0], "string")
+			}
+			str := string(dstr)
+
+			start, ok := args[1].(DInt)
+			if !ok {
+				return DNull, argTypeError(args[1], "int")
+			}
+
+			s := int(start)
+			if s < 0 || s > len(str)-1 {
+				return DNull, invalidArgError(fmt.Sprintf("%d is out of bouds", s))
+			}
+
+			count, ok := args[2].(DInt)
+			if !ok {
+				return DNull, argTypeError(args[2], "int")
+			}
+
+			c := int(count)
+			if c < 0 || s+c > len(str) {
+				return DNull, invalidArgError(fmt.Sprintf("%d is out of bouds", c))
+			}
+
+			return DString(str[s : s+c]), nil
+		},
+	},
+
+	"concat": {
+		nArgs: -1,
+		fn: func(args DTuple) (Datum, error) {
+			var buffer bytes.Buffer
+			for _, d := range args {
+				ds, error := datumToRawString(d)
+				if error != nil {
```

Comment:
mysql returns NULL if any argument is NULL. postgres ignored NULL arguments. We should pick one and document it.